### ISSUE TITLE
How to modify the structure to fit more than three channels of input pictures？

### ISSUE THREAD

**Issue by pyh99715** [VainF/DeepLabV3Plus-Pytorch#6]
The work I am currently facing needs to add a mask as a four-channel input based on the three-channel picture. I do n’t know how to change the network structure. For example, when using resnet101 as the backbone,how to modify the network structure to fit the four-channel  Picture input?
Hope for your help， Thanks

**Comment by VainF** [comment]
Hi @pyh99715. In ResNet101, _model.backbone.conv1_ is the first conv layer. You can build the model directly and then replace _model.backbone.conv1_ with a new 4-channel Conv layers, i.e., _nn.Conv2d(4, ...)_.

**Comment by pyh99715** [comment]
I have tried to find the first layer of the network structure before(resnet101 for example), but I can't seem to find the location that defines the first layer, is it in network/backbone/resnet.py line 48?

**Comment by VainF** [comment]
The first layer is defined in [line 139](https://github.com/VainF/DeepLabV3Plus-Pytorch/blob/master/network/backbone/resnet.py#L139) of ResNet.

**Comment by pyh99715** [comment]
I'm very sorry, I found it two minutes after I replied to you just now. I looked for it in the code of _deeplab.py for a long time. I am not familiar with this aspect.  💇‍♂ Then if resnet101 can take this form to change to four-channel input, others such as resnet50, mobilenet, etc. can also modify conv1 in this way to modify the entire network ?

**Comment by VainF** [comment]
Yes, you can change the first layer to match your inputs. 
But, the operation for mobilenet is more complicated because it does not have a separate conv1. You need to modify the [first element (L119)](https://github.com/VainF/DeepLabV3Plus-Pytorch/blob/master/network/backbone/mobilenetv2.py#L119) of model.backbone.features .

**Comment by pyh99715** [comment]
Thank you so much for your help, I even emailed you for help.  I think this project is not the same as many similar projects. It is more complete and efficient. May I ask if you implemented it yourself?
This is really an admirable job 👍 👍 👍
